bond/bond.go

```diff
@@ -102,18 +103,23 @@ func checkLinkExists(linkName string, netNsHandle *netlink.Handle) (netlink.Link
 }
 
 // configure the bonded link & add it using the netNsHandle context to add it to the required namespace. return a bondLinkObj pointer & error
-func createBondedLink(bondName string, bondMode string, bondMiimon string, failOverMac int, netNsHandle *netlink.Handle) (*netlink.Bond, error) {
+func createBondedLink(bondName string, bondMode string, bondMiimon string, bondMTU string, failOverMac int, netNsHandle *netlink.Handle) (*netlink.Bond, error) {
 	var err error
 	bondLinkObj := netlink.NewLinkBond(netlink.NewLinkAttrs())
 	bondModeObj := netlink.StringToBondMode(bondMode)
 	bondLinkObj.Attrs().Name = bondName
 	bondLinkObj.Mode = bondModeObj
 	bondLinkObj.Miimon, err = strconv.Atoi(bondMiimon)
-	bondLinkObj.FailOverMac = netlink.BondFailOverMac(failOverMac)
-
 	if err != nil {
 		return nil, fmt.Errorf("Failed to convert bondMiimon value (%+v) to an int, error: %+v", bondMiimon, err)
 	}
+	if len(bondMTU) > 0 {
```

Comment:
should we also validate that the MTU number provided is valid?

Reply:
Added validation that the mtu is not higher than the mtu of the pf of any of the vf's used. The link mac is used to match the pf's vf to the slave used in the bond.
The original vf mtu's are not relevant, as adding a link to a bond will modify it's mtu to that of the bond.